### ISSUE TITLE
customPostInitScripts doesn't run as root

### ISSUE THREAD

**Issue by spencerdcarlson** [bitnami/charts#17447]
### Name and Version

bitnami/wordpress:16.1.21

### What architecture are you using?

amd64

### What steps will reproduce the bug?

## Environment
```console
$ sw_vers
ProductName:		macOS
ProductVersion:		13.4.1
BuildVersion:		22F82
```

## Install 
```console
$ helm install -f values.yaml wp oci://registry-1.docker.io/bitnamicharts/wordpress
```

### Are you using any custom parameters or values?

### values.yaml
```yaml
volumePermissions:
  containerSecurityContext:
    runAsUser: 0
customPostInitScripts:
  custom-post-init.sh: |
    #!/bin/bash
    echo "Running custom post init script"
    whoami
    # apt -qq update && apt -yq install curl
...wordpress*parmeters 
```

### What is the expected behavior?

`whoami` should output root so I can execute root commands


### What do you see instead?

The init script is still being executed as the default non root user 1001
```
| wordpress 01:57:18.04 INFO  ==> Executing /docker-entrypoint-init.d/..2023_07_03_01_56_21.3667843249/custom-post-init.sh                                                                                                                                   │
│ Running custom post init script                                                                                                                                                                                                                            │
│ whoami: cannot find name for user ID 1001                                                                                                                                                                                                                  │
│ wordpress 01:57:18.05 ERROR ==> Failed to execute /docker-entrypoint-init.d/..2023_07_03_01_56_21.3667843249/custom-post-init.sh
```

### Additional information

Exact Images from describe

```console
$ kubectl describe pod wp-wordpress-84998d6866-svx65 | grep -B3 Image
Init Containers:
  volume-permissions:
    Container ID:  containerd://540d9f6c9508bd4140bcfe3dca2aee23d6dd8fb70705713707c1407d685bba24
    Image:         docker.io/bitnami/bitnami-shell:11-debian-11-r130
    Image ID:      docker.io/bitnami/bitnami-shell@sha256:98bba13bc2deed7c807da3a39d278b6d9b6cba8d906fb033d81240fc69ede83c
--
Containers:
  wordpress:
    Container ID:   containerd://f498a397e0acd74dca49c5083ebc6c9f588a84b78dbba0347528bd685ffb1c62
    Image:          docker.io/bitnami/wordpress:6.2.2-debian-11-r21
    Image ID:       docker.io/bitnami/wordpress@sha256:495fdde52fa8b58dc37ecdda8318f4985a04fb2afb8e345b6bb69fa547061944
```

**Comment by carrodher** [comment]
Please note you're changing the user for the _Volume Permission_ container but not for the main WordPress container which is the one in charge of running that script.
Could you try
```yaml
containerSecurityContext:
  runAsUser: 0
```
instead of
```yaml
volumePermissions:
  containerSecurityContext:
    runAsUser: 0
```
?

**Comment by spencerdcarlson** [comment]
Thanks for the prompt response. Will do. I believe I tried this as well and had the same result.

**Comment by spencerdcarlson** [comment]
> Could you try
> 
> ```yaml
> containerSecurityContext:
>   runAsUser: 0
> ```

I got a CreateContainerConfigError. 
```bash
$ kubectl describe pod wp-wordpress-6875654bf-v8mqj
 Warning  Failed                  6s (x9 over 83s)   kubelet                  Error: container's runAsUser breaks non-root policy (pod: "wp-wordpress-6875654bf-v8mqj_default(87153087-80f1-4cac-a6af-82628a54327f)", container: wordpress)
 ```
 So I included `runAsNonRoot`
 ```yaml
 containerSecurityContext:
  runAsNonRoot: false
  runAsUser: 0
  ```
  
  Which then gives me
  ```bash
|   wordpress 16:26:02.29                                                                                                                                                                                                                                      │
│ wordpress 16:26:02.29 Welcome to the Bitnami wordpress container                                                                                                                                                                                           │
│ wordpress 16:26:02.29 Subscribe to project updates by watching https://github.com/bitnami/containers                                                                                                                                                       │
│ wordpress 16:26:02.29 Submit issues and feature requests at https://github.com/bitnami/containers/issues                                                                                                                                                   │
│ wordpress 16:26:02.29                                                                                                                                                                                                                                      │
│ wordpress 16:26:02.29 INFO  ==> ** Starting WordPress setup **                                                                                                                                                                                             │
│ wordpress 16:26:02.31 INFO  ==> Generating sample certificates                                                                                                                                                                                             │
│ Generating RSA private key, 4096 bit long modulus (2 primes)                                                                                                                                                                                               │
│ ........................++++                                                                                                                                                                                                                               │
│ ...................................................................................................................................++++                                                                                                                    │
│ e is 65537 (0x010001)                                                                                                                                                                                                                                      │
│ Signature ok                                                                                                                                                                                                                                               │
│ subject=CN = example.com                                                                                                                                                                                                                                   │
│ Getting Private key                                                                                                                                                                                                                                        │
│ realpath: /bitnami/apache/conf: No such file or directory                                                                                                                                                                                                  │
│ wordpress 16:26:02.83 INFO  ==> Configuring the HTTP port                                                                                                                                                                                                  │
│ wordpress 16:26:02.84 INFO  ==> Configuring the HTTPS port                                                                                                                                                                                                 │
│ wordpress 16:26:02.84 INFO  ==> Configuring Apache ServerTokens directive                                                                                                                                                                                  │
│ chown: changing ownership of '/opt/bitnami/php/var/run/session': Operation not permitted                                                                                                                                                                   │
│ chown: changing ownership of '/opt/bitnami/php/var/run': Operation not permitted                                                                                                                                                                           │
│ Stream closed EOF for default/wp-wordpress-5f5ccddb96-9d8nj (wordpress)
```

**Comment by carrodher** [comment]
Probably you will need to enable the `volumePermission` init container as well, this container changes the owner/group of the PV mount point to `runAsUser:fsGroup`

```yaml
containerSecurityContext:
 runAsNonRoot: false
 runAsUser: 0
volumePermissions:
  enabled: true
```
